account-kit/signer/src/client/base.ts

```diff
@@ -315,22 +315,34 @@ export abstract class BaseSignerClient<TExportWalletParams = unknown> {
    * that result here.
    *
    * @param {Hex} msg the hex representation of the bytes to sign
+   * @param {string} mode specify if signing should happen for solana or ethereum
    * @returns {Promise<Hex>} the signature over the raw hex
    */
-  public signRawMessage = async (msg: Hex): Promise<Hex> => {
+  public signRawMessage = async (
+    msg: Hex,
+    mode: "SOLANA" | "ETHEREUM" = "ETHEREUM"
+  ): Promise<Hex> => {
     if (!this.user) {
       throw new NotAuthenticatedError();
     }
 
+    if (!this.user.solanaAddress && mode === "SOLANA") {
+      throw new Error("No Solana address available for the user");
```

Comment:
TODO: we probably want login to create solana addresses for legacy users. This will need signer service support which can be tacked on easily

Reply:
not handling this here